qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,324 @@
+    """A ground state calculation employing the AdaptVQE algorithm.
+
+    The performance of AdaptVQE can significantly depend on the choice of gradient method, QFI
+    solver (if applicable) and the epsilon value.
+    """
+
+    def __init__(
+        self,
+        solver: VQE ,
+        ansatz: Optional[QuantumCircuit] = None,
+        threshold: float = 1e-5,
+        max_iterations: Optional[int] = None,
+        adapt_gradient: Optional[GradientBase] = None,
+        initial_point: Optional[np.ndarray] = None,
+        expectation: Optional[ExpectationBase] = None,
+        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
+        excitation_pool: List[Union[OperatorBase, QuantumCircuit]] = None,
+    ) -> None:
+        """
+        Args:
+            ansatz: A parameterized circuit used as Ansatz for the wave function.
+            solver: a factory for the VQE solver employing a UCCSD ansatz.
+            threshold: the energy convergence threshold. It has a minimum value of 1e-15.
+            max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
+            initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+            excitation_pool: An entire list of excitations.
+        """
+        validate_min("threshold", threshold, 1e-15)
+        super().__init__(
+            initial_point = initial_point
+        )
+
+        if adapt_gradient is None:
+            adapt_gradient = Gradient(grad_method="param_shift")
+        self._threshold = threshold
+        self.solver = solver
+        self._max_iterations = max_iterations
+        self._adapt_gradient = adapt_gradient
+        self._excitation_pool = excitation_pool
+        self._tmp_ansatz = ansatz
+        self.expectation = expectation
+        self.quantum_instance = quantum_instance
+        self._excitation_list: List[OperatorBase] = []
+
+    def _compute_gradients(
+        self,
+        theta: List[float],
+        operator: OperatorBase,
+    ) -> List[Tuple[float, PauliSumOp]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters
+            vqe: The variational quantum eigensolver instance used for solving
+
+        Returns:
+            List of pairs consisting of gradient and excitation operator.
+        """
+        res = []
+        # compute gradients for all excitation in operator pool
+        sampler = CircuitSampler(self.quantum_instance)
+        for exc in self._excitation_pool:
+            # add next excitation to ansatz
+            self._tmp_ansatz.operators = self._excitation_list + [exc]
+            # the ansatz needs to be decomposed for the gradient to work
+            self.ansatz = self._tmp_ansatz.decompose()
```

Comment:
You need to replace this everywhere:
```suggestion
            self.solver.ansatz = self._tmp_ansatz.decompose()
```